client/src/app/pages/dashboard/profile/profile-run-history/profile-run-history.component.ts

```diff
@@ -64,4 +78,18 @@ export class ProfileRunHistoryComponent implements OnInit {
     this.currentPage = pageNum;
     this.loadRunHistory();
   }
+
+  onFilterApply() {
+    // Some destructuring to shorten the upcoming if statement
+    const { isPersonalBest, map, order } = this.filterFG.value;
+    if ( // Don't do anything if the filters didn't change
+      this.currentFilter.isPersonalBest === isPersonalBest ||
+      this.currentFilter.map === map ||
+      this.currentFilter.order === order
+    )
+      return;
```

Comment:
This logic doesn't seem correct. If you change one but not the other(s), wouldn't this return true and not search with the new query?

Reply:
Which in this case you'd just want to change the `||` to `&&` to verify *nothing* changed, as opposed to *something* which it currently checks for.

Reply:
Forgot to flip `||` and `&&' when I flipped the boolean logic. Fixed in 97e96fe